### PR TITLE
feat(route): 添加bilibili会员购票务路由(#13618)

### LINKED ISSUE #13618: bilibili会员购票务
### 类型

社交媒体

### 网站地址

https://show.bilibili.com/platform/home.html

### 网站描述

票务信息

### 需要生成什么内容？

票务上新

### 额外描述

按城市、类型 设置路由或参数

### 这不是重复的 RSS 请求

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues) 和 [pull requests](https://github.com/DIYgod/RSSHub/pulls)，以确保该 RSS 尚未被请求。

### PR DESCRIPTION
**Pull request by nightmare-mio**
## Involved Issue / 该 PR 相关 Issue

Close #13618

## Example for the Proposed Route(s) / 路由地址示例

```routes
/bilibili/platform/-1/全部类型
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [X] New Route / 新的路由
  - [X] Follows [v2 Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [v2 路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [X] Documentation / 文档说明
  - [ ] EN / 英文文档
  - [X] CN / 中文文档
- [ ] Full text / 全文获取
  - [ ] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制 
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施? 
- [X] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [X] Parsed / 可以解析
  - [X] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/v2/bilibili/maintainer.js [hunk 1]
```diff
@@ -42,4 +42,5 @@ module.exports = {
     '/vsearch/:kw/:order?/:disableEmbed?/:tid?': ['Symty'],
     '/watchlater/:uid/:disableEmbed?': ['JimenezLi'],
     '/weekly/:disableEmbed?': ['ttttmr'],
+    '/platform/:area?/:p_type?/:uid?': ['nightmare-mio'],
```
Comment (TonyRL) [inline lib/v2/bilibili/maintainer.js:45]: Please insert new entries in [alphabetical order](https://docs.rsshub.app/joinus/advanced/script-standard#general-guidelines).

File: lib/v2/bilibili/router.js [hunk 2]
```diff
@@ -42,4 +42,5 @@ module.exports = (router) => {
     router.get('/vsearch/:kw/:order?/:disableEmbed?/:tid?', require('./vsearch'));
     router.get('/watchlater/:uid/:disableEmbed?', require('./watchlater'));
     router.get('/weekly/:disableEmbed?', require('./weekly_recommend'));
+    router.get('/platform/:area?/:p_type?/:uid?', require('./platform'));
```
Comment (TonyRL) [inline lib/v2/bilibili/router.js:45]: Please insert new entries in alphabetical order.

File: website/docs/routes/social-media.mdx [hunk 3]
```diff
@@ -341,6 +341,15 @@ UP 主关注用户现在需要 b 站登录后的 Cookie 值，所以只能自建
...
 <Route author="DIYgod" example="/bilibili/mall/ip/0_3000294" path="/bilibili/mall/ip/:id" paramsDesc={['作品 id, 可在作品列表页 URL 中找到']}/>
 
+### 会员购票务 {#bilibili-hui-yuan-gou-piao-wu}
+
+<Route author="nightmare-mio" example="http://127.0.0.1:1200/bilibili/platform/" path="/platform/:area?/:p_type?/:uid?" paramsDesc={['省市-国标码,默认->-1即全国','类型：见下表，默认->全部类型','用户id，可以不填，不过不填不设置cookie，搜索结果与登入账号后搜索结果不一样。可以在url中找到，需要配置cookie值，只需要SESSDATA的值即可']}/>
+
+| 类型 |
+| 演出 |
+| 展览 |
+| 本地生活 |
```
Comment (TonyRL) [inline website/docs/routes/social-media.mdx:351]: You need hyphens for [Markdown table](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-tables).
```suggestion
| 类型 |
| ---- |
| 演出 |
| 展览 |
| 本地生活 |
```

File: website/docs/routes/social-media.mdx [hunk 4]
```diff
@@ -341,6 +341,15 @@ UP 主关注用户现在需要 b 站登录后的 Cookie 值，所以只能自建
 
 <Route author="DIYgod" example="/bilibili/mall/ip/0_3000294" path="/bilibili/mall/ip/:id" paramsDesc={['作品 id, 可在作品列表页 URL 中找到']}/>
 
+### 会员购票务 {#bilibili-hui-yuan-gou-piao-wu}
+
+<Route author="nightmare-mio" example="http://127.0.0.1:1200/bilibili/platform/" path="/platform/:area?/:p_type?/:uid?" paramsDesc={['省市-国标码,默认->-1即全国','类型：见下表，默认->全部类型','用户id，可以不填，不过不填不设置cookie，搜索结果与登入账号后搜索结果不一样。可以在url中找到，需要配置cookie值，只需要SESSDATA的值即可']}/>
```
Comment (TonyRL) [inline website/docs/routes/social-media.mdx:346]: `example` should not incldude server address.
`path` should [include the namespace](https://docs.rsshub.app/joinus/new-rss/add-docs) `bilibili`

File: lib/v2/bilibili/radar.js [hunk 5]
```diff
@@ -143,5 +143,13 @@ module.exports = {
                 target: '/bilibili/live/room/:roomID',
             },
         ],
+        show: [
+            {
+                title: 'bilibili会员购票务',
+                docs: 'https://docs.rsshub.app/routes/social-media#bilibili-hui-yuan-gou-zuo-pin',
+                source: '/platform/:area?/:p_type?/:uid?',
+                target: '/platform/:area?/:p_type?/:uid?',
```
Comment (TonyRL) [inline lib/v2/bilibili/radar.js:151]: `target` should start with the namespace.

File: lib/v2/bilibili/platform.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        const coordinate = `<div>活动地点: ${item.city} ${item.venue_name}</div>`;
+        const live_time = `<div>活动时间: ${item.tlabel}</div>`;
+        const staff = `<div>参展览嘉宾: ${item.staff}</div>`;
+        const countdown = `<div>结束日期: ${item.countdown}</div>`;
+        const price = `<div>最低价: ${item.price_low / 100} ; 最高价: ${item.price_high / 100}</div>`;
+        return {
+            title: item.project_name,
+            link: item.url,
+            description: body_html + coordinate + live_time + staff + countdown + price,
+            pubDate: parseDate(new Date(item.sale_start_time * 1000)),
```
Comment (TonyRL) [inline lib/v2/bilibili/platform.js:45]:
```suggestion
            pubDate: parseDate(item.sale_start_time * 1000),
```

File: lib/v2/bilibili/platform.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+    const headers =
+        cookie === undefined
+            ? {
+                  Referer: `https://space.bilibili.com`,
+              }
+            : {
+                  Referer: `https://space.bilibili.com`,
+                  Cookie: `SESSDATA=${cookie}`,
+              };
```
Comment (TonyRL) [inline lib/v2/bilibili/platform.js:24]:
```suggestion
    const headers = {
        Referer: 'https://space.bilibili.com',
        Cookie: cookie ? `SESSDATA=${cookie}` : undefined,
    };
```

File: lib/v2/bilibili/platform.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+ * @Date: 2023-11-24 22:51:19
+ * @LastEditTime: 2023-11-25 01:05:33
+ * @Description:
+ */
+const got = require('@/utils/got');
+const { parseDate } = require('@/utils/parse-date');
+const config = require('@/config').value;
+
+module.exports = async (ctx) => {
+    const { area = -1, p_type = '全部类型', uid } = ctx.params;
```
Comment (TonyRL) [inline lib/v2/bilibili/platform.js:12]: [`camelCase`](https://docs.rsshub.app/joinus/advanced/script-standard#naming)

File: lib/v2/bilibili/platform.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        url: `${link}?version=134&page=1&pagesize=16&area=${area}&filter=&platform=web&p_type=${p_type}`,
+        headers,
+    });
+    // 列表
+    const list = response.data.result;
+
+    const items = list.map((item) => {
+        const body_html = `<img src="${item.cover}"/>`;
+        const coordinate = `<div>活动地点: ${item.city} ${item.venue_name}</div>`;
+        const live_time = `<div>活动时间: ${item.tlabel}</div>`;
```
Comment (TonyRL) [inline lib/v2/bilibili/platform.js:37]: `camelCase`

File: lib/v2/bilibili/radar.js [hunk 10]
```diff
@@ -143,5 +143,13 @@ module.exports = {
                 target: '/bilibili/live/room/:roomID',
             },
         ],
+        show: [
+            {
+                title: 'bilibili会员购票务',
```
Comment (TonyRL) [inline lib/v2/bilibili/radar.js:148]: [Do not](https://docs.rsshub.app/joinus/new-radar#title) repeat the website name (`_name`).
```suggestion
                title: '会员购票务',
```

File: lib/v2/bilibili/platform.js (hunk 9 above)
Comment (nightmare-mio) [reply lib/v2/bilibili/platform.js]: i think that i fixed this，i changed body_html to bodyHtml.I hope what I understand is correct.

File: lib/v2/bilibili/router.js [hunk 11]
```diff
@@ -1,3 +1,9 @@
+/*
+ * @Author: nightmare-mio wanglongwei2009@qq.com
+ * @Date: 2023-11-24 22:22:32
+ * @LastEditTime: 2023-11-25 08:18:42
+ * @Description:
+ */
```
Comment (TonyRL) [inline lib/v2/bilibili/router.js:6]:
```suggestion
```

File: website/docs/routes/social-media.mdx [hunk 12]
```diff
@@ -341,6 +341,16 @@ UP 主关注用户现在需要 b 站登录后的 Cookie 值，所以只能自建
 
 <Route author="DIYgod" example="/bilibili/mall/ip/0_3000294" path="/bilibili/mall/ip/:id" paramsDesc={['作品 id, 可在作品列表页 URL 中找到']}/>
 
+### 会员购票务 {#bilibili-hui-yuan-gou-piao-wu}
+
+<Route author="nightmare-mio" example="/bilibili/platform/-1/全部类型" path="/platform/:area?/:p_type?/:uid?" paramsDesc={['省市-国标码,默认->-1即全国','类型：见下表，默认->全部类型','用户id，可以不填，不过不填不设置cookie，搜索结果与登入账号后搜索结果不一样。可以在url中找到，需要配置cookie值，只需要SESSDATA的值即可']}/>
```
Comment (TonyRL) [inline website/docs/routes/social-media.mdx:346]: https://github.com/DIYgod/RSSHub/pull/13880#discussion_r1404562174
```suggestion
<Route author="nightmare-mio" example="/bilibili/platform/-1/全部类型" path="/bilibili/platform/:area?/:p_type?/:uid?" paramsDesc={['省市-国标码,默认->-1即全国','类型：见下表，默认->全部类型','用户id，可以不填，不过不填不设置cookie，搜索结果与登入账号后搜索结果不一样。可以在url中找到，需要配置cookie值，只需要SESSDATA的值即可']}/>
```

File: website/docs/routes/social-media.mdx [hunk 13]
```diff
@@ -341,6 +341,16 @@ UP 主关注用户现在需要 b 站登录后的 Cookie 值，所以只能自建
 
 <Route author="DIYgod" example="/bilibili/mall/ip/0_3000294" path="/bilibili/mall/ip/:id" paramsDesc={['作品 id, 可在作品列表页 URL 中找到']}/>
 
+### 会员购票务 {#bilibili-hui-yuan-gou-piao-wu}
+
+<Route author="nightmare-mio" example="/bilibili/platform/-1/全部类型" path="/bilibili/platform/:area?/:p_type?/:uid?" paramsDesc={['省市-国标码,默认->-1即全国','类型：见下表，默认->全部类型','用户id，可以不填，不过不填不设置cookie，搜索结果与登入账号后搜索结果不一样。可以在url中找到，需要配置cookie值，只需要SESSDATA的值即可']}/>
```
Comment (TonyRL) [inline website/docs/routes/social-media.mdx:346]:
```suggestion
<Route author="nightmare-mio" example="/bilibili/platform/-1/全部类型" path="/bilibili/platform/:area?/:p_type?/:uid?" paramsDesc={['省市-国标码,默认为-1即全国','类型：见下表，默认为全部类型','用户id，可以不填，不过不填不设置cookie，搜索结果与登入账号后搜索结果不一样。可以在url中找到，需要配置cookie值，只需要SESSDATA的值即可']}/>
```

File: lib/v2/bilibili/router.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (nightmare-mio) [reply lib/v2/bilibili/router.js:6]: i'm sorry,due to my plugin.

**Pull request merged**